assets/js/phoenix_live_view/rendered.js

```diff
@@ -205,55 +265,14 @@ export default class Rendered {
 
   recursiveCIDToString(components, cid, onlyCids, allowRootComments = true){
     let component = components[cid] || logError(`no component for CID ${cid}`, components)
-    let template = document.createElement("template")
     let [html, streams] = this.recursiveToString(component, components, onlyCids)
-    template.innerHTML = html
-    let container = template.content
     let skip = onlyCids && !onlyCids.has(cid)
+    let attrs = {[PHX_COMPONENT]: cid, id: `${this.parentViewId()}-${cid}`}
```

Comment:
Could we use the new phx-magic-id already and battle test it within components? This way we don't need to mangle the ID either in `modifyRoot`.